salt/metalk8s/salt/master/certs/salt-api.sls

```diff
@@ -0,0 +1,33 @@
+{%- from "metalk8s/map.jinja" import etcd with context %}
+
+include:
+  - metalk8s.internal.m2crypto
+
+Create Salt API private key:
+  x509.private_key_managed:
+    - name: /etc/kubernetes/pki/salt-api.key
+    - bits: 2048
+    - verbose: False
+    - user: root
+    - group: root
+    - mode: 600
+    - makedirs: True
+    - dir_mode: 755
+    - require:
+      - metalk8s_package_manager: Install m2crypto
+
+Generate Salt API certificate:
+  x509.certificate_managed:
+    - name: /etc/kubernetes/pki/salt-api.crt
+    - public_key: /etc/kubernetes/pki/salt-api.key
+    - ca_server: {{ pillar['metalk8s']['ca']['minion'] }}
+    - signing_policy: {{ etcd.cert.apiserver_client_signing_policy }}
```

Comment:
You should use this signing policy ` kube_api.cert.server_signing_policy` as it's a server auth certs IMO

Reply:
You also need to add some subjectAltName or change the `CN`.
For me we want at least our control_plane ip and fqdn, hostname (maybe `localhost/127.0.0.1` but I don't think so) as subjectAltName.
Maybe @NicolasT has a precise idea for this one

Reply:
Meh, too much copy-pasta :sweat_smile:

Reply:
As @TeddyAndrieux already mentioned, indeed, this should be a server policy being applied, otherwise the cert won't even work.

As for the CN, let's not simply use `salt-api`, but a more descriptive name, e.g. `salt-api on {{ grains.id }}` or something along those lines.

Then, we indeed should set up a proper list of `subjectAltName`s. This should include
- For *external* access:
  * The control-plane IP of the host, `grains['metalk8s']['control_plane_ip']`
  * `grains['fqdn']`
- For *internal* access: the DNS names of the Salt service (when running in the cluster), with various suffices:
  * `salt-master`
  * `salt-master.kube-system`
  * `salt-master.kube-system.svc`
  * `salt-master.kube-system.svc.cluster.local`

Reply:
Thanks for the info.

Done.